src/data/list/alist.lean

```diff
@@ -241,6 +245,47 @@ ext $ by simp only [alist.insert_entries, list.kerase_cons_eq, and_self, alist.s
 theorem to_alist_cons (a : α) (b : β a) (xs : list (sigma β)) :
   list.to_alist (⟨a,b⟩ :: xs) = insert a b xs.to_alist := rfl
 
+theorem mk_cons_eq_insert (c : sigma β) (l : list (sigma β)) (h : (c :: l).nodupkeys) :
+  (⟨c :: l, h⟩ : alist β) = insert c.1 c.2 ⟨l, nodupkeys_of_nodupkeys_cons h⟩ :=
+by simpa [insert] using (kerase_of_not_mem_keys $ not_mem_keys_of_nodupkeys_cons h).symm
+
+/-- Recursion on an `alist`, using `insert`. -/
+@[elab_as_eliminator] def recursion {β : α → Type v} {C : alist β → Sort*} (H0 : C ∅)
+  (IH : Π (a : α) (b : β a) (l : alist β) (h : a ∉ l), C l → C (l.insert a b)) : Π l : alist β, C l
```

Comment:
Does it work with the `induction` tactic? I saw a PR recently that changed the order of arguments in a custom recursor to make it work with `induction`. Could you please add an `example` or a test showing that it does?

What about `insert_rec` as a name (not sure)? Compare with `finset.cons_induction`.

Reply:
I've added an example, and renamed to `insert_rec`.